File: dotnet/MyOrganization/MyOrganization.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MyOrganization
{
    internal class MyOrganization : Organization
    {
        override protected Position CreateOrganization()
        {
            Position ceo = new Position("CEO");
            Position pres = new Position("President");
            ceo.AddDirectReport(pres);
            Position vpm = new Position("VP Marketing");
            pres.AddDirectReport(vpm);
            Position vps = new Position("VP Sales");
            pres.AddDirectReport(vps);
            Position vpf = new Position("VP Finance");
            pres.AddDirectReport(vpf);
            Position coo = new Position("COO");
            pres.AddDirectReport(coo);
            Position cio = new Position("CIO");
            ceo.AddDirectReport(cio);
            Position vpt = new Position("VP Technology");
            cio.AddDirectReport(vpt);
            Position vpi = new Position("VP Infrastructure");
            cio.AddDirectReport(vpi);
            Position dea = new Position("Director Enterprise Architecture");
            vpt.AddDirectReport(dea);
            Position dct = new Position("Director Customer Technology");
            vpt.AddDirectReport(dct);
            Position s = new Position("Salesperson");
            vps.AddDirectReport(s);

            return ceo;
        }

        static void Main()
        {
            MyOrganization org = new MyOrganization();
            org.Hire(new Name("Bob", "Smith"), "CEO");
            org.Hire(new Name("Zaphod", "Beeblebrox"), "President");
            org.Hire(new Name("Gandalf", "Gray"), "Director Enterprise Architecture");
            org.Hire(new Name("Bill", "Lumbergh"), "Director Customer Technology");
            org.Hire(new Name("Ford", "Prefect"), "VP Marketing");
            org.Hire(new Name("Jane", "Seller"), "VP Sales");
            org.Hire(new Name("Bean", "Counter"), "VP Finance");
            org.Hire(new Name("Victoria", "Sinclair"), "CIO");
            org.Hire(new Name("Head", "Geek"), "VP Technology");
            org.Hire(new Name("Steve", "Dent"), "VP Infrastructure");
            org.Hire(new Name("Slick", "Willie"), "Salesperson");
            Console.WriteLine(org);
        }
    }
}

File: typescript/MyOrganization.ts
import Name from './Name';
import Organization from './Organization';
import Position from './Position';

class MyOrganization extends Organization {
  createOrganization(): Position {
    const ceo = new Position('CEO');
    const pres = new Position('President');
    ceo.addDirectReport(pres);
    const vpm = new Position('VP Marketing');
    pres.addDirectReport(vpm);
    const vps = new Position('VP Sales');
    pres.addDirectReport(vps);
    const vpf = new Position('VP Finance');
    pres.addDirectReport(vpf);
    const coo = new Position('COO');
    pres.addDirectReport(coo);
    const cio = new Position('CIO');
    ceo.addDirectReport(cio);
    const vpt = new Position('VP Technology');
    cio.addDirectReport(vpt);
    const vpi = new Position('VP Infrastructure');
    cio.addDirectReport(vpi);
    const dea = new Position('Director Enterprise Architecture');
    vpt.addDirectReport(dea);
    const dct = new Position('Director Customer Technology');
    vpt.addDirectReport(dct);
    const s = new Position('Salesperson');
    vps.addDirectReport(s);
    return ceo;
  }
}

const myOrg = new MyOrganization();
myOrg.hire(new Name('Bob', 'Smith'), 'CEO');
myOrg.hire(new Name('Zaphod', 'Beeblebrox'), 'President');
myOrg.hire(new Name('Gandalf', 'Gray'), 'Director Enterprise Architecture');
myOrg.hire(new Name('Bill', 'Lumbergh'), 'Director Customer Technology');
myOrg.hire(new Name('Ford', 'Prefect'), 'VP Marketing');
myOrg.hire(new Name('Jane', 'Seller'), 'VP Sales');
myOrg.hire(new Name('Bean', 'Counter'), 'VP Finance');
myOrg.hire(new Name('Victoria', 'Sinclair'), 'CIO');
myOrg.hire(new Name('Head', 'Geek'), 'VP Technology');
myOrg.hire(new Name('Steve', 'Dent'), 'VP Infrastructure');
myOrg.hire(new Name('Slick', 'Willie'), 'Salesperson');
console.log(myOrg.toString());
